llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -40,21 +41,11 @@ class ComposerHFCausalLM(HuggingFaceModelWithZLoss):
                 to validation metrics. Default: ``False``.
     """
 
-    def __init__(self,
-                 om_model_config: DictConfig,
-                 om_tokenizer_config: Optional[DictConfig] = None):
+    def __init__(self, om_model_config: DictConfig, tokenizer: Tokenizer):
```

Comment:
we went from being "optional" to not being optional

I think the tokenizer is only needed to get vocab_size (ie len(tokenizer)); this is probably part of the model (emb layer dim)
Otherwise, tokenizer is passed to `HuggingFaceModelWithZLoss` which is a subclass of HuggingFaceModel from Composer, where the tokenizer input is optional ([see here](https://github.com/mosaicml/composer/blob/052446000033d9026adabf8087f8e1e5d0e47abd/composer/models/huggingface.py#L67))

Reply:
Good point, let me fix this. I can just read the `vocab_size` from the `hf_config` correct?

Reply:
Actually, I'm not sure if the `Optional` from previous code was actually optional. If `om_tokenizer_config = None`, then `resolved_om_tokenizer_config['name']` would fail. I think the `type: ignore` is masking a bug there.

OK to merge?

Reply:
yes merge.